skynet/dgpu/compute.py

```diff
@@ -81,153 +69,140 @@ def prepare_params_for_diffuse(
         _params
     )
 
+_model_name: str = ''
+_model_mode: str = ''
+_model = None
 
-# TODO, yet again - drop the redundant prefix ;)
-class SkynetMM:
-    '''
-    (AI algo) Model manager for loading models, computing outputs,
-    checking load state, and unloading when no-longer-needed/finished.
+@cm
+def maybe_load_model(name: str, mode: str):
+    if mode == 'diffuse':
+        mode = 'txt2img'
 
-    '''
-    def __init__(self, config: dict):
-        self.cache_dir = None
-        if 'hf_home' in config:
-            self.cache_dir = config['hf_home']
+    global _model_name, _model_mode, _model
+    config = load_skynet_toml().dgpu
 
-        self._model_name: str = ''
-        self._model_mode: str = ''
+    if _model_name != name or _model_mode != mode:
```

Comment:
personal opi but, i tend to prefer the multiline style for chained logic expressions for a few reasons,

1. they're easier to mutate later,
1. they more clearly (especially with the use of `()`) distinguish the sub-expression patterning which can be a lot nicer when you're trying to get the predicate output *jist* just from a glance.
1. you can more easily embed comment lines around sub-expressions to explain them.

so in this case i'd have put,

```python
if (
    _model_name != name
    or
    _model_mode != mode
):
    ...

Reply:
Agreed! done in `structify`